network/network.go

```diff
@@ -676,57 +569,17 @@ func (n *network) Disconnected(nodeID ids.NodeID) {
 	}
 }
 
-func (n *network) Peers(peerID ids.NodeID) ([]ips.ClaimedIPPort, error) {
-	// Only select validators that we haven't already sent to this peer
-	unknownValidators, ok := n.gossipTracker.GetUnknown(peerID)
-	if !ok {
-		n.peerConfig.Log.Debug(
-			"unable to find peer to gossip to",
-			zap.Stringer("nodeID", peerID),
-		)
-		return nil, nil
-	}
-
-	// We select a random sample of validators to gossip to avoid starving out a
-	// validator from being gossiped for an extended period of time.
-	s := sampler.NewUniform()
-	s.Initialize(uint64(len(unknownValidators)))
-
-	// Calculate the unknown information we need to send to this peer.
-	validatorIPs := make([]ips.ClaimedIPPort, 0, int(n.config.PeerListNumValidatorIPs))
-	for i := 0; i < len(unknownValidators) && len(validatorIPs) < int(n.config.PeerListNumValidatorIPs); i++ {
-		drawn, err := s.Next()
-		if err != nil {
-			return nil, err
-		}
-
-		validator := unknownValidators[drawn]
-		n.peersLock.RLock()
-		_, isConnected := n.connectedPeers.GetByID(validator.NodeID)
-		peerIP := n.peerIPs[validator.NodeID]
-		n.peersLock.RUnlock()
-		if !isConnected {
-			n.peerConfig.Log.Verbo(
-				"unable to find validator in connected peers",
-				zap.Stringer("nodeID", validator.NodeID),
-			)
-			continue
-		}
-
-		// Note: peerIP isn't used directly here because the TxID may be
-		//       incorrect.
-		validatorIPs = append(validatorIPs,
-			ips.ClaimedIPPort{
-				Cert:      peerIP.Cert,
-				IPPort:    peerIP.IPPort,
-				Timestamp: peerIP.Timestamp,
-				Signature: peerIP.Signature,
-				TxID:      validator.TxID,
-			},
-		)
-	}
+func (n *network) KnownPeers() ([]byte, ids.ID) {
+	return n.validatorTracker.Bloom()
+}
 
-	return validatorIPs, nil
+func (n *network) Peers(except ids.NodeID, knownPeers *bloom.ReadFilter, salt ids.ID) []*ips.ClaimedIPPort {
```

Comment:
Do we need to rate limit this like pull gossip per nodeID?

Reply:
I suppose if they want to use their bandwidth allocation doing this, that's fine?